dev/infra/Dockerfile

```diff
@@ -91,10 +91,10 @@ RUN mkdir -p /usr/local/pypy/pypy3.8 && \
     ln -sf /usr/local/pypy/pypy3.8/bin/pypy /usr/local/bin/pypy3.8 && \
     ln -sf /usr/local/pypy/pypy3.8/bin/pypy /usr/local/bin/pypy3
 RUN curl -sS https://bootstrap.pypa.io/get-pip.py | pypy3
-RUN pypy3 -m pip install numpy 'six==1.16.0' 'pandas<=2.1.4' scipy coverage matplotlib lxml
+RUN pypy3 -m pip install numpy 'six==1.16.0' 'pandas<=2.2.0' scipy coverage matplotlib lxml
```

Comment:
shall we use `>=` to confirm 2.2.0+ is used?

Reply:
or using `==`? just because `<=` cannot confirm 2.2.0 is used

Reply:
AFAIK, pip automatically finds the most recent version that meets the conditions as below (Has this not worked well so far btw??):

```
(pyspark-dev-env) spark % pip install "pandas<=2.2.0"
Collecting pandas<=2.2.0
...
Installing collected packages: pandas
Successfully installed pandas-2.2.0
```

But I'm okay with the way `==`. WDYT, @HyukjinKwon ?

Reply:
Let's pin this to 2.2.0 for now. I think we have seen some issues when automaticaly using latest pandas version

Reply:
Got it. btw Pandas 2.2.0 again introduces some breaking changes 😂 Let me address it